experiments/sample_ssl4eo.py

```diff
@@ -0,0 +1,193 @@
+# Radius (meter) is half the wanted patch size.
+
+"""
+
+import argparse
+import csv
+import os
+import time
+import warnings
+from typing import Any, Dict, List, Tuple
+
+import numpy as np
+from rtree import index
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+warnings.simplefilter("ignore", UserWarning)
+
+
+""" samplers to get locations of interest points"""
+
+
+def get_world_cities(download_root: str = "world_cities") -> List[Dict[str, Any]]:
+    url = "https://simplemaps.com/static/data/world-cities/basic/simplemaps_worldcities_basicv1.71.zip"  # noqa: E501
+    filename = "worldcities.csv"
+    if not os.path.exists(os.path.join(download_root, os.path.basename(url))):
+        download_and_extract_archive(url, download_root)
+    with open(os.path.join(download_root, filename), encoding="UTF-8") as csvfile:
+        reader = csv.DictReader(csvfile, delimiter=",", quotechar='"')
+        cities = []
+        for row in reader:
+            row["population"] = (
+                row["population"].replace(".", "") if row["population"] else "0"
+            )
+            cities.append(row)
+    return cities
+
+
+def get_interest_points(
+    cities: List[Dict[str, str]], size: int = 10000
+) -> List[List[float]]:
+    cities = sorted(cities, key=lambda c: int(c["population"]), reverse=True)[:size]
+    points = [[float(c["lng"]), float(c["lat"])] for c in cities]
+    return points
+
+
+def km2deg(kms: float, radius: float = 6371) -> float:
+    return kms / (2.0 * radius * np.pi / 360.0)
+
+
+def sample_point(interest_points: List[List[float]], std: float) -> Tuple[float, float]:
+    rng = np.random.default_rng()
+    point = rng.choice(interest_points)
+    std = km2deg(std)
+    lon, lat = np.random.normal(loc=point, scale=[std, std])
+    return (lon, lat)
+
+
+def create_bbox(
+    coords: Tuple[float, float], bbox_size_degree: float
+) -> Tuple[float, float, float, float]:
+    lon, lat = coords
+    bbox = (
+        lon - bbox_size_degree,
+        lat - bbox_size_degree,
+        lon + bbox_size_degree,
+        lat + bbox_size_degree,
+    )
+    return bbox
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--save-path", type=str, default="./data/", help="dir to save data"
+    )
+    parser.add_argument(
+        "--radius", type=float, default=1320, help="patch radius in meters"
+    )
+    parser.add_argument(
+        "--overlap-ratio",
+        type=float,
+        default=0.25,
+        help="max overlap ratio between adjacent patches",
+    )
+    parser.add_argument(
+        "--num-cities", type=int, default=10000, help="number of cities to sample"
+    )
+    parser.add_argument(
+        "--std", type=int, default=50, help="std of gaussian distribution"
+    )
+    parser.add_argument(
+        "--resume", type=str, default=None, help="resume from a previous run"
```

Comment:
Why not make this a boolean and instead use `--save-path` for the file to resume from?

Reply:
some freedom to load from a customize-named file?